app/(app)/offence/vehicle.tsx

```diff
@@ -30,11 +34,27 @@ const Page = () => {
     }
   }, [data, setOffenceState])
 
+  useEffect(() => {
+    return navigation.addListener('beforeRemove', (e) => {
+      e.preventDefault()
+
+      if (e.data.action.type === 'GO_BACK') {
+        setOffenceState({ vehicleId: undefined })
```

Comment:
What happens if the vehilcle is already undefined? should we not let the user go back then?